core/src/main/java/dev/streamx/cli/ingestion/BasePublishUnpublishCommand.java

```diff
@@ -0,0 +1,74 @@
+package dev.streamx.cli.ingestion;
+
+import dev.streamx.cli.exception.IngestionClientException;
+import dev.streamx.clients.ingestion.StreamxClient;
+import dev.streamx.clients.ingestion.exceptions.StreamxClientException;
+import jakarta.inject.Inject;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Spec;
+
+public abstract class BasePublishUnpublishCommand implements Runnable {
+
+  @ArgGroup(exclusive = false)
+  protected IngestionArguments ingestionArguments;
+
+  @Spec
+  protected CommandSpec spec;
+
+  @Inject
+  protected SchemaProvider schemaProvider;
+
+  @Inject
+  protected StreamxClientProvider streamxClientProvider;
+
+  protected abstract void performCommand(StreamxClient client) throws StreamxClientException;
+
+  @Override
+  public void run() {
+    StreamxClient client = createClient();
+
+    try {
+      performCommand(client);
+    } catch (StreamxClientException e) {
+      if (is404Error(e)) {
+        createExtendedIngestionClientException(e);
+      } else {
+        createIngestionClientException(e);
+      }
+    } finally {
+      try {
+        client.close();
+      } catch (StreamxClientException e) {
+        throw new IngestionClientException(e);
+      }
+    }
+  }
+
+  private StreamxClient createClient() {
+    try {
+      return streamxClientProvider.createStreamxClient();
+    } catch (StreamxClientException e) {
+      throw new IngestionClientException(e);
```

Comment:
why rethrow ingestionClientException ? Its problem with streamxClient not ingestion

Reply:
Because in previous code, it worked that that, so I didn't want to change the behavior.

Creating streamxClient was part of a try/catch block, that threw IngestionClientException in case of any StreamxClientException